app/helpers/application_helper.rb

```diff
@@ -3,4 +3,20 @@ module ApplicationHelper
   def validate_params
     params[:donation][:amount].to_i > 0
   end
+
+  def bootstrap_class_for flash_type
+    { success: "alert-success", errors: "alert-danger", alert: "alert-warning", notice: "alert-info" }[flash_type.to_sym] || flash_type.to_s
+  end
+
+
+
+  def flash_messages(opts = {})
+    flash.each do |msg_type, message|
+      concat(content_tag(:div, sanitize(message), class: "alert #{bootstrap_class_for(msg_type)} fade in") do
+              concat content_tag(:button, 'x', class: "close", data: { dismiss: 'alert' })
+              concat message
+            end)
```

Comment:
`end` at 18, 12 is not aligned with `content_tag(:div, sanitize(message), class: "alert #{bootstrap_class_for(msg_type)} fade in") do` at 15, 13 or `concat(content_tag(:div, sanitize(message), class: "alert #{bootstrap_class_for(msg_type)} fade in") do` at 15, 6